cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -274,6 +270,10 @@ protected TestManager getTestManager() {
     }
   }
 
+  protected RouteConfigManager getRouteConfigManager(NamespaceId namespaceId, String appName, String serviceName) {
+    return new RemoteRouteConfigManager(namespaceId, appName, serviceName, getClientConfig(), getRestClient());
```

Comment:
so this is wrong, right? 

The method says it will return RouteConfigManager, but it always returns RemoteRouteConfigManager. So technically, you can even call ``getRouteConfigManager`` in unit tests, but it will return the remote implementation? Shouldn't we let the ``TestManager`` class decide which implementation to use based on the runtime?